### PR TITLE
Comma-separated list of data

### PR DESCRIPTION
**Pull request by ChristofferL**
Added functionality to be able to get a comma-separated list of all stories (/comma-separated-data), and choose which fields to include by passing url-arguments

### CODE REVIEW

File: src/main/java/com/sonymobile/backlogtool/HomeController.java [hunk 1]
```diff
@@ -622,6 +621,103 @@ public ModelAndView themeepic(
...
+        if (fields != null && !fields.isEmpty()) {
+            StringBuilder sb = new StringBuilder();
+            Session session = sessionFactory.openSession();
+            Transaction tx = null;
+            try {
+                tx = session.beginTransaction();
+
+                String queryString = null;
+                if (archived == null) {
+                    queryString = "select distinct s from Story s";
```
Comment (fredriklindman) [inline src/main/java/com/sonymobile/backlogtool/HomeController.java]: is distinct needed here?

File: src/main/java/com/sonymobile/backlogtool/HomeController.java [hunk 2]
```diff
@@ -622,6 +621,103 @@ public ModelAndView themeepic(
...
+                } else {
+                    queryString = "select distinct s from Story s "
```
Comment (fredriklindman) [inline src/main/java/com/sonymobile/backlogtool/HomeController.java]: and here?

File: src/main/java/com/sonymobile/backlogtool/HomeController.java [hunk 3]
```diff
@@ -622,6 +621,103 @@ public ModelAndView themeepic(
...
+                        } else if (field.equals("contributorsite")) {
+                            sb.append(s.getContributorSite());
+                        } else if (field.equals("datearchived")) {
+                            if (s.getDateArchived() != null) {
+                                sb.append(sdf.format(s.getDateArchived()));
+                            }
+                        } else if (field.equals("storyattr1")) {
+                            AttributeOption attr = s.getStoryAttr1();
+                            String attrStr = "";
+                            if(attr != null) {
```
Comment (fredriklindman) [inline src/main/java/com/sonymobile/backlogtool/HomeController.java]: format, and same below

File: src/main/webapp/WEB-INF/views/comma-separated-data.jsp [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+                <li>storyattr1</li>
+                <li>storyattr2</li>
+                <li>storyattr3</li>
+                <li>area</li>
+                <li>deadline</li>
+                <li>description</li>
+            </ul>
+        </p>
+        <p> For example: </p>
+        <p class="link-box"><a href="?fields=id,title,storyattr1,deadline">?fields=id,title,storyattr1,deadline&archived=true</a></p>
```
Comment (fredriklindman) [inline src/main/webapp/WEB-INF/views/comma-separated-data.jsp]: maybe you could change this into the example from the feature request to save them some time every month ;)

**Comment by fredriklindman** [comment]
Overall it looks great, just some small comments :+1:

**Comment by ChristofferL** [comment]
Fixes according to comments.

**Comment by ChristofferL** [comment]
Added error-message-handling for invalid fields

**Pull request merged**